src/modules/signer/walletConnectMetamask.ts

```diff
@@ -12,6 +12,8 @@ export const fromWalletConnectMetamask = async (bridge: string, infuraId?: strin
     await walletProvider.enable();
     const provider = new providers.Web3Provider(walletProvider);
     const signerService = new SignerService(provider.getSigner(), ProviderType.MetaMask);
```

Comment:
This code seems weird. The provider is coming from `createWalletConnectProvider` but `SignerService` is initialized with `ProviderType.MetaMask`? Is this correct?

Reply:
thanks for noticing. may be this ProviderType isn't needed really. it is only used to register listeners of connected provider, but it wouldn't harm if listeners of other providers was registered

Reply:
> it is only used to register listeners of connected provider
I don't understand. What is the relationship of this class to WalletConnect or Metamask?

Reply:
This class is notified by wallet to clean session or reinitialize itself when account is changed
 https://github.com/energywebfoundation/iam-client-lib/blob/44e0ad3ecb8c3fecc98bd717e389bbb60cfbbe2e/src/modules/signer/signer.service.ts#L78